src/ui/handlers/abstract-ui-handler.ts

```diff
@@ -18,25 +36,133 @@ export abstract class UiHandler {
     this.mode = mode;
   }
 
-  abstract setup(): void;
+  public get active(): boolean {
+    return this._active;
+  }
+
+  public get ready(): boolean {
+    return this._ready;
+  }
+
+  /**
+   * Prepares the handler for display.
+   * Should not be overridden. Calls {@linkcode setup}, which subclasses should override.
+   */
+  public initialize(): void {
+    if (this._ready) {
+      console.warn("Attempting to initialize an already ready handler. Aborting.");
+      return;
+    }
+
+    this.setup();
+    this._ready = true;
+  }
+
+  /**
+   * Destroys the handler and makes it no longer usable.
+   * Should not be overridden. Calls {@linkcode tearDown}, which subclasses should override.
+   */
+  public destroy(): void {
+    if (!this.ready) {
+      console.warn("Attempting to destroy a non initialized handler. Aborting.");
+      return;
+    }
+
+    // If the handler is active, stop it first.
+    if (this.active) {
+      this.stop();
+    }
+
+    this.tearDown();
+    this._ready = false;
+  }
+
+  /**
+   * Displays the handler.
+   * Should not be overridden. Calls {@linkcode show}, which subclasses should override.
+   *
+   * @param args the arguments needed by this handler for display.
```

Comment:
NIT:  Recently learned that the dash `-` is supposed to be put manually.
```suggestion
   * @param args - The arguments needed by this handler for display.
```